doc: update testing components doc with generated compileComponent (#41947)
angular/angular: aio/content/guide/testing-components-scenarios.md

To use `waitForAsync()` functionality, you must import `zone.js/testing` in your test setup file.
If you created your project with the Angular CLI, `zone-testing` is already imported in `src/test.ts`.

Here's the previous `fakeAsync()` test, re-written with the `waitForAsync()` utility.

<code-example
  path="testing/src/app/twain/twain.component.spec.ts"
  region="waitForAsync-test">
</code-example>

> The `async()` utility hides some asynchronous boilerplate by arranging for the tester's code

Reviewer: ```suggestion
The `waitForAsync()` utility hides some asynchronous boilerplate by arranging for the tester's code
```

<div class="alert is-critical">

If you neglect to make the test function async

> (e.g., forget to use `async()` as described below),

Reviewer: ```suggestion
(e.g., forget to use the `async` keyword as described below),
```

1.  An async `beforeEach()` that compiles the components
1.  A synchronous `beforeEach()` that performs the remaining setup.

> To follow this pattern, import the `async()` helper with the other testing symbols.

Reviewer: This is no longer a helper, it is a language construct...

```suggestion
To follow this pattern, make the function asynchronous by adding the `async` keyword.
```

  region="async-before-each"
  header="app/banner/banner-external.component.spec.ts (async beforeEach)"></code-example>

> The `async()` helper function takes a parameterless function with the body of the setup.

Reviewer: I think we should just remove this sentence.
